File: 01_gpio_digital/led_blink_task.c
#include <wiringPi.h>
#include <stdio.h>
#define LED_RED 4
#define LED_GRE 17
#define LED_YEL 22
int main (void)
{
  // wiringPiSetup () ;
  wiringPiSetupGpio();
  pinMode (LED_RED, OUTPUT) ; // OUTPUT
  pinMode (LED_GRE, OUTPUT) ;
  pinMode (LED_YEL, OUTPUT) ;
  // LED ON OFF
  digitalWrite (LED_RED, HIGH) ; delay (2000) ;
  digitalWrite (LED_RED,  LOW) ;
  digitalWrite (LED_GRE, HIGH) ; delay (2000) ;
  digitalWrite (LED_GRE,  LOW) ;
  digitalWrite (LED_YEL, HIGH) ; delay (2000) ;
  digitalWrite (LED_YEL,  LOW) ;
  
  pinMode (LED_RED, INPUT) ; // INPUT
  pinMode (LED_GRE, INPUT) ;
  pinMode (LED_YEL, INPUT) ;
  
  return 0 ;
}

File: 01_gpio_digital/led_blink_task.py
import RPi.GPIO as GPIO
import time

RED_LED = 17
YEL_LED = 27
GRE_LED = 22

GPIO.setmode(GPIO.BCM)  # GPIO.BCM or GPIO.BOARD
GPIO.setup(RED_LED, GPIO.OUT)  # GPIO.OUT or GPIO.IN
GPIO.setup(YEL_LED, GPIO.OUT)  # GPIO.OUT or GPIO.IN
GPIO.setup(GRE_LED, GPIO.OUT)  # GPIO.OUT or GPIO.IN

# def blink(red, yellow, green):
    

GPIO.output(RED_LED, GPIO.HIGH) # LED_RED 1
print("RED LED ON")
time.sleep(2)
GPIO.output(RED_LED, GPIO.LOW) # LED_RED 0
print("RED LED OFF")
GPIO.output(YEL_LED, GPIO.HIGH) # LED_YELLOW 1
print("YELLOW LED ON")
time.sleep(2)
GPIO.output(YEL_LED, GPIO.LOW) # LED_YELLOW 0
print("YELLOW LED OFF")
GPIO.output(GRE_LED, GPIO.HIGH) # LED_GREEN 1
print("GREEN LED ON")
time.sleep(2)
GPIO.output(GRE_LED, GPIO.LOW) # LED_GREEN 0
print("GREEN LED OFF")

GPIO.cleanup()  # GPIO 핀 상태 초기화
